src/main/java/core/basesyntax/strategy/BalanceHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Inventory;
+
+public class BalanceHandler implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Inventory inventory) {
+        inventory.getInventory().put(transaction.getFruit(),
+                Math.max(transaction.getQuantity(), 0));
```

Comment:
The use of `Math.max(transaction.getQuantity(), 0)` ensures that negative quantities are not set in the inventory, which is good. However, the 'BALANCE' operation should set the quantity to the exact value provided in the transaction, without the need for the `Math.max` check, as negative quantities should be checked and handled before this point.